perf/fateman.jl

```diff
@@ -10,9 +10,9 @@ function fateman1(degree::Int)
     s = TaylorN( [oneH, HomogeneousPolynomial([one(T),one(T),one(T),one(T)],1)], degree )
     s = s^degree
     # s is converted to order 2*ndeg
-    s = TaylorN(s, 2*degree)
+    s = TaylorN(s.coeffs, 2*degree)
```

Comment:
The reason is related to a method I deleted of `TaylorN` (as well as the corresponding for `Taylor1` and `HomogeneousPolynomial`). The deleted method (in constructors.jl) was `TaylorN{T<:Number}(x::TaylorN{T}, order::Int) = TaylorN{T}(x.coeffs, order)`. Those methods have side effects, which may yield some unexpected behaviour...